tests/layer_tests/tensorflow_tests/test_tf_SegmentMax.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2018-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+
+rng = np.random.default_rng()
+
+class TestSegmentMax(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        assert 'data:0' in inputs_info
+        assert 'segment_ids:0' in inputs_info
+        data_shape = inputs_info['data:0']
+        segment_ids_shape = inputs_info['segment_ids:0']
+        inputs_data = {}
+        inputs_data['data:0'] = rng.integers(-50, 50, data_shape).astype(np.float32)
```

Comment:
please test with different data type

Reply:
Now there are tests for float32, float64 and int32.